crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -676,40 +678,64 @@ where
                 value,
                 ..
             }) => {
-                // If we're in a class or module scope, then the annotation needs to be
-                // available at runtime.
-                // See: https://docs.python.org/3/reference/simple_stmts.html#annotated-assignment-statements
-                let runtime_annotation = if self.semantic.future_annotations() {
-                    self.semantic
+                enum AnnotationKind {
+                    RuntimeRequired,
+                    RuntimeEvaluated,
+                    TypingOnly,
```

Comment:
I would find some documenting of the kinds similar to what you wrote in the PR summary useful to better understand the code.

Reply:
Good call, done.